File: evidence/generated-systems/golang/fintech-on-golang/fintech-on-golang/internal/model/PaymentProcessor.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// PaymentProcessor Declaration
//==============================================================
type PaymentProcessor struct {
    gorm.Model
     Name                                    string
    ProcessorCode                                    string
    NetworkSupport                                    string
     Institutions           []FinancialInstitution `gorm:"foreignKey:InstitutionsFromPaymentProcessorId"`
     Contracts           []PaymentContract `gorm:"foreignKey:ContractsFromPaymentProcessorId"`
     Settlements           []SettlementBatch `gorm:"foreignKey:SettlementsFromPaymentProcessorId"`

// parent associations as their child

}

File: evidence/generated-systems/angular/fintech-on-angular/src/app/models/PaymentProcessor.js

// Define collection and schema for PaymentProcessor
export interface PaymentProcessor {
    name:
	type : string
    processorCode:
	type : string
    networkSupport:
	type : string
    Institutions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FinancialInstitution' }]
    Contracts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'PaymentContract' }]
    Settlements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'SettlementBatch' }]
#
    collection: 'paymentProcessors'
}
